### PR TITLE
Move Plan and Task domain classes to OpenSRP Plan Evaluator

### PR DESCRIPTION
**Pull request by rehammuzzamil**
Issue # 10
- Move Plan and Task domain and their dependent classes to opensrp plan evaluator library.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -87,6 +88,32 @@
 			<artifactId>fhir-path</artifactId>
 			<version>${fhir.version}</version>
 		</dependency>
+
+		<dependency>
+			<groupId>org.projectlombok</groupId>
```
Comment (rkodev) [inline pom.xml:93]: Are we still using lombok on this lib?

Comment (rehammuzzamil) [reply pom.xml:93]: @rkodev are we suppose to remove this?

Comment (githengi) [reply pom.xml:93]: let me confirm this will work on androiud

Comment (githengi) [reply pom.xml:93]: Lombok works on android. We should be using it

File: src/main/java/org/smartregister/domain/Condition.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@ToString
+public class Condition {
```
Comment (githengi) [inline src/main/java/org/smartregister/domain/Condition.java:14]: Maybe add Builder annotation

File: src/main/java/org/smartregister/domain/Expression.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@ToString
+public class Expression {
```
Comment (githengi) [inline src/main/java/org/smartregister/domain/Expression.java:14]: Maybe add Builder annotation

File: src/main/java/org/smartregister/domain/Expression.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    private String name;
+    private String language;
+    private String expression;
```
Comment (githengi) [inline src/main/java/org/smartregister/domain/Expression.java:17]: Add `SubjectConcept` as property of this class

File: src/main/java/org/smartregister/domain/Trigger.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@ToString
+public class Trigger {
```
Comment (githengi) [inline src/main/java/org/smartregister/domain/Trigger.java:14]: maybe add a @Builder annotation as well

**Pull request merged**